From 338b2590a049a02715c7797552580b0c58ec8e8d Mon Sep 17 00:00:00 2001
From: demostanis <demostanis@protonmail.com>
Date: Mon, 17 Nov 2025 11:22:46 +0100
Subject: [PATCH] s/Gio.DesktopAppInfo/GioUnix/DesktopAppInfo/

Gio.DesktopAppInfo is deprecated
(and apparently lua-lgi doesn't include deprecated variables??)
---
 widget/app_launcher/init.lua | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/widget/app_launcher/init.lua b/widget/app_launcher/init.lua
index 8e48b0e..a51f62a 100755
--- a/widget/app_launcher/init.lua
+++ b/widget/app_launcher/init.lua
@@ -1,4 +1,5 @@
 local Gio = require("lgi").Gio
+local GioUnix = require("lgi").GioUnix
 local awful = require("awful")
 local gobject = require("gears.object")
 local gtable = require("gears.table")
@@ -626,9 +627,9 @@ local function generate_apps(self)
                         end
                     end
 
-                    local desktop_app_info = Gio.DesktopAppInfo.new(app_info.get_id(app))
-                    local terminal = Gio.DesktopAppInfo.get_string(desktop_app_info, "Terminal") == "true" and true or false
-                    local generic_name = Gio.DesktopAppInfo.get_string(desktop_app_info, "GenericName") or nil
+                    local desktop_app_info = GioUnix.DesktopAppInfo.new(app_info.get_id(app))
+                    local terminal = GioUnix.DesktopAppInfo.get_string(desktop_app_info, "Terminal") == "true" and true or false
+                    local generic_name = GioUnix.DesktopAppInfo.get_string(desktop_app_info, "GenericName") or nil
 
                     table.insert(self._private.all_entries, {
                         name = name,
-- 
2.50.0

